### DISCUSSION TITLE
get aktual adress using Teslalogger API

### DISCUSSION THREAD

**Discussion by adnet-solutions** [bassmaster187/TeslaLogger#1060] [Q&A]
Hi,
is it possible somehow to get the actual Adress from the CAR from Teslalogger? It show the MAP and location on a map, but can i get the adress as JASON somehow?

**Comment by bassmaster187** [comment]
Actually we query the Nominatim service from Open Street Map just for parking stops. The Nominatim Usage Policy allows just a maximum of 1 request per second. So that’s not possible. I have an own Nominatim Server and could offer you what you want, but not for free because it draws a lot of traffic on my server. I think 5$ or 5€ / Month should cover my cost. 

 

https://operations.osmfoundation.org/policies/nominatim/

 

Von: bicmac ***@***.***> 
Gesendet: Mittwoch, 14. Juni 2023 15:08
An: bassmaster187/TeslaLogger ***@***.***>
Cc: Subscribed ***@***.***>
Betreff: [bassmaster187/TeslaLogger] get aktual adress using Teslalogger API (Discussion #1060)

 

Hi,
is it possible somehow to get the actual Adress from the CAR from Teslalogger? It show the MAP and location on a map, but can i get the adress as JASON somehow?

**Reply by adnet-solutions** [reply]
Hi @bassmaster187 in which format tyour server will give me the address. i tryed once a Nomination request and the JSON had different fields every time for coordinates. Sometimes i got city, sometime only suburb or town.
Orblem is that the fields not empty they just not exsists if a city is not there. Any idea how to fix that?

**Reply by superfloh247** [reply]
like this https://github.com/bassmaster187/TeslaLogger/blob/4fa25a8e68feab7c9738306105a2021a98b5f102/TeslaLogger/WebHelper.cs#L3599
